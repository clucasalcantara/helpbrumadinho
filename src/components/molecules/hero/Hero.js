import React from 'react'
import { string } from 'prop-types'
import SearchBar from '../search-bar'

const Hero = ({ headline, placeholder }) => (
  <section className="ui vertical center aligned segment hb-hero">
    <div className="ui text container">
      <h1 className="ui header">{headline}</h1>
      <SearchBar placeholder={placeholder} />
    </div>
  </section>
)

Hero.propTypes = {
  headline: string
}

Hero.defaultProps = {
  headline: 'Headline'
}

export default Hero
