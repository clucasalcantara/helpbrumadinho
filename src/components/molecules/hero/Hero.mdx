---
name: Hero
menu: Molecules
---

import { Playground, PropsTable } from 'docz'
import Hero from './Hero'

# Hero component

## Basic usage

<Playground>
  <Hero />
</Playground>

## Properties

<PropsTable of={Hero} />
