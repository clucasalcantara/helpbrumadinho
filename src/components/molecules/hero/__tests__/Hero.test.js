import React from 'react'
import { shallow } from 'enzyme'
import Hero from '../Hero'

describe('Hero component', () => {
  it('should renders without crash', () => {
    const wrapper = shallow(<Hero headline="Work with coffee" />)

    expect(wrapper).toMatchSnapshot()
  })
})
