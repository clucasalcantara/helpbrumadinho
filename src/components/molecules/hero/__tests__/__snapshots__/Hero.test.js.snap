// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero
    headline="Work with coffee"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ui text container"
      >
        <h1
          className="ui header"
        >
          Work with coffee
        </h1>
        <SearchBar
          handleSelect={[Function]}
        />
      </div>,
      "className": "ui vertical center aligned segment hb-hero",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="ui header"
          >
            Work with coffee
          </h1>,
          <SearchBar
            handleSelect={[Function]}
          />,
        ],
        "className": "ui text container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Work with coffee",
            "className": "ui header",
          },
          "ref": null,
          "rendered": "Work with coffee",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSelect": [Function],
            "placeholder": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ui text container"
        >
          <h1
            className="ui header"
          >
            Work with coffee
          </h1>
          <SearchBar
            handleSelect={[Function]}
          />
        </div>,
        "className": "ui vertical center aligned segment hb-hero",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="ui header"
            >
              Work with coffee
            </h1>,
            <SearchBar
              handleSelect={[Function]}
            />,
          ],
          "className": "ui text container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Work with coffee",
              "className": "ui header",
            },
            "ref": null,
            "rendered": "Work with coffee",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleSelect": [Function],
              "placeholder": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
